packages/rest/src/__tests__/integration/rest.application.integration.ts

```diff
@@ -145,8 +145,44 @@ describe('RestApplication (integration)', () => {
     });
   });
 
+  it('controls redirect routes', async () => {
+    givenApplication();
+    restApp.controller(DummyController);
+    restApp.redirect('/page/html', '/html');
+    await restApp.start();
+    client = createRestAppClient(restApp);
+    const response = await client.get('/page/html').expect(303);
+    await client.get(response.header.location).expect(200, 'Hi');
+  });
+
+  it('controls redirect routes', async () => {
```

Comment:
Please use unique test names that describe the scenario being tested: what are the conditions, what is being done, what is the expected result.

I am confused about these two tests. What are you trying to cover and how are these two tests different from the acceptance-level tests added earlier?

Reply:
this is clearly code duplication .. tests replaced with custom status code .

Reply:
>     * What happens when `basePath` is set - how is the target URL constructed. Must have: server-level test. Nice to have: acceptance-level test against an applicaition.

i think we have enough test that covers all the redirect angel . theirs no need to check the test base path if it is set . in server and acceptance test i make a new request to check location and the location is based on base path . two birds one stone 
lets assume some one delete base path from redirect method all redirect test that depends on base path will fail .

Reply:
> two birds one stone

In my experience, it's better to have more smaller tests where each test is focused on a single scenario. See https://loopback.io/doc/en/lb4/Defining-your-testing-strategy.html